v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DataStreamToSpannerLT100Gb.java

```diff
@@ -0,0 +1,194 @@
+
+    String dlqGcsPrefix =
+        String.join("/", new String[] {TEST_ROOT_DIR, gcsResourceManager.runId(), testName, "dlq"});
+    SubscriptionName dlqSubscription =
+        createPubsubResources(
+            TEST_ROOT_DIR + testName + "dlq",
+            pubsubResourceManager,
+            dlqGcsPrefix,
+            gcsResourceManager);
+
+    // Setup Datastream
+    MySQLSource mySQLSource = getMySQLSource();
+    Stream stream =
+        createDatastreamResources(
+            ARTIFACT_BUCKET, gcsPrefix, mySQLSource, datastreamResourceManager);
+
+    // Setup Parameters
+    Map<String, String> params =
+        new HashMap<>() {
+          {
+            put("inputFilePattern", getGcsPath(ARTIFACT_BUCKET, gcsPrefix));
+            put("streamName", stream.getName());
+            put("instanceId", spannerResourceManager.getInstanceId());
+            put("databaseId", spannerResourceManager.getDatabaseId());
+            put("projectId", project);
+            put("deadLetterQueueDirectory", getGcsPath(ARTIFACT_BUCKET, dlqGcsPrefix));
+            put("gcsPubSubSubscription", subscription.toString());
+            put("dlqGcsPubSubSubscription", dlqSubscription.toString());
+            put("datastreamSourceType", "mysql");
+            put("inputFileFormat", "avro");
+          }
+        };
+
+    LaunchConfig.Builder options = LaunchConfig.builder(getClass().getSimpleName(), SPEC_PATH);
+    options.addEnvironment("maxWorkers", MAX_WORKERS).addEnvironment("numWorkers", NUM_WORKERS);
+
+    options.setParameters(params);
+
+    // Act
+    PipelineLauncher.LaunchInfo jobInfo = pipelineLauncher.launch(project, region, options.build());
+    assertThatPipeline(jobInfo).isRunning();
+
+    List<ConditionCheck> checks = new ArrayList<>();
+    for (int i = 1; i <= 10; ++i) {
+      checks.add(
+          SpannerRowsCheck.builder(spannerResourceManager, TABLE + i)
+              .setMinRows(6500000)
+              .setMaxRows(6500000)
+              .build());
+    }
+
+    ChainedConditionCheck conditionCheck = ChainedConditionCheck.builder(checks).build();
+
+    PipelineOperator.Result result =
+        pipelineOperator.waitForCondition(
+            createConfig(jobInfo, Duration.ofHours(4), Duration.ofMinutes(5)), conditionCheck);
+
+    // Assert Conditions
+    assertThatResult(result).meetsConditions();
+
+    result = pipelineOperator.cancelJobAndFinish(createConfig(jobInfo, Duration.ofMinutes(5)));
+    assertThatResult(result).isLaunchFinished();
+
+    // export results
+    exportMetricsToBigQuery(jobInfo, getMetrics(jobInfo));
+  }
+
+  public MySQLSource getMySQLSource() {
+    String password =
+        secretClient.accessSecret(
+            "projects/269744978479/secrets/nokill-datastream-mysql-to-spanner-password/versions/1");
+    MySQLSource mySQLSource =
+        new MySQLSource.Builder("34.41.46.219", "root", password, 3306).build();
```

Comment:
Please get the IP's also from secrets. 
We should not be checking in credentials on public repos

Reply:
Done.